src/cascadia/TerminalApp/MinMaxCloseControl.cpp

```diff
@@ -17,59 +17,26 @@ namespace winrt::TerminalApp::implementation
         winrt::Windows::UI::Xaml::Application::LoadComponent(*this, resourceLocator, winrt::Windows::UI::Xaml::Controls::Primitives::ComponentResourceLocation::Nested);
     }
 
-    uint64_t MinMaxCloseControl::ParentWindowHandle() const
+    void MinMaxCloseControl::Maximize()
     {
-        return reinterpret_cast<uint64_t>(_window);
+        winrt::Windows::UI::Xaml::VisualStateManager::GoToState(MaximizeButton(), L"WindowStateMaximized", false);
     }
 
-    void MinMaxCloseControl::ParentWindowHandle(uint64_t handle)
+    void MinMaxCloseControl::RestoreDown()
     {
-        _window = reinterpret_cast<HWND>(handle);
+        winrt::Windows::UI::Xaml::VisualStateManager::GoToState(MaximizeButton(), L"WindowStateNormal", false);
     }
 
-    void MinMaxCloseControl::_OnMaximize(byte flag)
+    // These need to be defined here to make the compiler happy. They do nothing

```

Comment:
This is only because they are defined in the .xaml file. Remove them there, and it removes them here.

Reply:
You should really expose events off MinMaxCloseControl and forward the button presses through them, so as to not expose its inner members

Reply:
That's what I originally wanted to do. Lemme futz with it some more.